integr8ly/integreatly-operator
# MGDAPI-4039: Removing RHMI for the code base

# Issue link
[https://issues.redhat.com/browse/MGDAPI-4039](https://issues.redhat.com/browse/MGDAPI-4039)

# What
This work is to remove the old RHMI product from the operator.
As RHMI has hit its EOL support date there is no need maintain or make business logic decisions based on RHMI.

# Verification steps

**NOTE:** Commits have rearranged, squashed and fixed up, checkout an individual commit does not mean the code will compile.

Open questions:
* Can the assets/icons folder be removed?

As there is a very large area to verify, the verification is broken out into the areas that will need to be reviewed.
The hope is to cover every area but assume that something has been missed from the list and keep an eye out for any kind of issue.

## Code Review
With the removal of the RHMI a lot of if-statements have being removed which was checking for the installation type.
Think about the effect the removal of the if-statements might have when looking at the sandbox deployments.

The rhmi-config has being removed.
CRO seemed to be the only area that had this dependency.
Check that this dependency was correctly refactored.

## RBAC
As the rhmiConfig CR has being removed from the installation, any RBAC rules relating to this should also have being removed.
Same goes for the products that RHMI installed.
Check for RBAC rules that are out of place with the current set of products being installed.
Our backlog does [MGDAPI-3974](https://issues.redhat.com/browse/MGDAPI-3974) which is about review the RBAC access levels.
In this PR we want to make sure that RBAC is removed for components that are no longer installed.

## Building Images
There has being a number of Dockerfiles that have had a change made to them.
Build all Dockerfiles found in the repo.
Use the make targets where possible. 

This testing should also include the building of indexes and bundles.

## unit tests
Refactoring of the unit test has happened.
Check that refactor test are still fit for propose.
Unit test should all pass.
This is possibly covered by the PROW check but no harm in a manual run being carried out.

## e2e tests
e2e test are ran by prow but this is not enough.
Prow uses a limited set of tests.
A jenkins pipeline needs to run the full suite.
The e2eosd is also going to be affected by this work, these test should also be checked.

## Releases
There has being updates done to the prepare release script. 
It would be good to validate the that bundles are generate correctly and bundles can be included in indexes.
Running the release pipeline would also be a good idea if the managed tenants repos can be mocked out.

## Deployments
### Local Development
Running the operator should work as normal.
The defaults for environment variables should now all be mapped to managed-api. 

Uninstall should work as normal.
The cleanup script should be un affected by the changes.

#### OLM Installations
For testing the OLM installations it should require the building of images for the operator, bundle and indexes.
This should be as close the addon flow that we can get. 

#### Installation
A fresh installation should be done.
Nothing is expected to be out of place. 
Configure the testing IDP and check that all routes still work as expected.
Configure a simple rated limited end point to ensure rate limiting is still functioning.

#### upgrade
Set up the cluster with the existing addon version of the operator.
Configure the testing IDP and rate limiting example. 
Upgrade should happen without any issue.

It would be a good idea to check the upgrade as both a service affecting and non-service affecting.
This would prove there has being no on wanted changes to the upgrade logic.

#### uninstall
For the uninstallation use the method that add the label to the config map. 
This method is as close to how hive will do this action.

## Dashboards, Alerts and Metrics
As all these are very much linked it would be good to review all at the same time.
Do an OLM installation as the metrics depend on having the operator running on the cluster.

Review all the dashboards, both internal and customer facing.
All panels in the dashboards should be working and not have any mention of RHMI

Same goes for the alerts.
There should be no alerts with the RHMI reference. 
Check metrics used in the queries to ensure these metrics are still exposed.

For the metrics, no metric should now have reference to RHMI.
